### ISSUE TITLE
Arithmetic overflow error converting numeric to data type numeric.

### ISSUE THREAD

**Issue by RudeySH** [MiniProfiler/dotnet#300]
I'm using the latest prerelease version of MiniProfiler v4, and I'm using SQL Server for storing profiler results.

One of the numeric fields is overflowing. I think it's `DurationMilliseconds`, because I'm profiling background jobs that can run for more than an hour.

What's the max duration supported when storing profiler results using `SqlServerStorage`? And can it be increased?

```
System.Data.SqlClient.SqlException (0x80131904): Arithmetic overflow error converting numeric to data type numeric.
The statement has been terminated.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Dapper.SqlMapper.ExecuteCommand(IDbConnection cnn, CommandDefinition& command, Action`2 paramReader) in C:\projects\dapper\Dapper\SqlMapper.cs:line 2827
   at Dapper.SqlMapper.ExecuteImpl(IDbConnection cnn, CommandDefinition& command) in C:\projects\dapper\Dapper\SqlMapper.cs:line 570
   at Dapper.SqlMapper.Execute(IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Nullable`1 commandTimeout, Nullable`1 commandType) in C:\projects\dapper\Dapper\SqlMapper.cs:line 443
   at StackExchange.Profiling.Storage.SqlServerStorage.Save(MiniProfiler profiler) in C:\projects\dotnet\src\MiniProfiler.Providers.SqlServer\SqlServerStorage.cs:line 77
   at StackExchange.Profiling.DefaultProfilerProvider.Stopped(MiniProfiler profiler, Boolean discardResults) in C:\projects\dotnet\src\MiniProfiler.Shared\ProfileProviders\DefaultProfilerProvider.cs:line 52
   at StackExchange.Profiling.MiniProfiler.Stop(Boolean discardResults) in C:\projects\dotnet\src\MiniProfiler.Shared\MiniProfiler.cs:line 224
```

**Comment by NickCraver** [comment]
I concur that's the likely column giving issues in your situation, however, addressing this would be a breaking change for existing users. Can you try expanding the column in your database? For example:
```sql
Alter Table MiniProfilers Alter Column DurationMilliseconds decimal(15,1) not null;
Alter Table MiniProfilerTimings Alter Column DurationMilliseconds decimal(15,3) not null;
```

If that resolves it, I'll look at changing the types in the creation script...but let's confirm it's the core issue first :)

**Comment by NickCraver** [comment]
@RudeySH any update here?

**Comment by SaltyDH** [comment]
Just hit this issue as well - I'm making these changes as we speak to see if they resolve the issue.

**Comment by SaltyDH** [comment]
Due to #310 

This patch needs to be applied to:

```
Alter Table MiniProfilers Alter Column DurationMilliseconds decimal(15,1) not null;
Alter Table MiniProfilerClientTimings Alter Column DurationMilliseconds decimal(15,3) not null;
```

**Comment by RudeySH** [comment]
I haven't seen the arithmetic overflow error after altering the columns. I just checked all logs from the past week.

**Comment by SaltyDH** [comment]
My 1h 12min process just finished without error - looks good.

**Comment by NickCraver** [comment]
Fixed in e42c0b5 for the next build, thanks for the help everyone!
